### PR TITLE
CommitsReporter: add support for per-commit-view

### PR DESCRIPTION
**Pull request by yamidark**
Updating the branch as we are working on #468

### CODE REVIEW

**Comment by yamidark** [comment]
The dashboard for this currently doesn't work as we have changed the backend `json` report files without the frontend portion yet.

@ongspxm Can double check the new `commits.json` format if it is what you wish to work with, or if any changes need to be made.

**Comment by yamidark** [comment]
>Looks fine to me, but why dont we continue using AuthorIntervalContribution.java and specify our own time / date range instead?

Hmmm, iirc, at that time after discussing with @ongspxm, we decided that the calculating of weekly interval contribution would be done in the frontend, so we went with this. If we plan to do shorter intervals, like for each 6 hours or so, we could also shift the calculation of that to the frontend, just we need to add the time as well to the `commit.json` report.

I also don't think we would want to store too much different intervals in the `commit.json` report, as it can get extremely large, which would make the loading of the page slower as well.

**Comment by ongspxm** [comment]
okay, the format is something i can work with

**Comment by ongspxm** [comment]
since its a side branch, im just gonna merge yah?

File: src/systemtest/resources/noDateRange/expected/reposense_testrepo-Alpha_master/commits.json [hunk 1]
```diff
@@ -1,141 +1,134 @@
 {
-  "authorWeeklyIntervalContributions": {
+  "authorDailyContributionsMap": {
     "fakeAuthor": [
       {
         "insertions": 0,
         "deletions": 0,
-        "sinceDate": "2018-02-06",
```
Comment (ongspxm) [inline src/systemtest/resources/noDateRange/expected/reposense_testrepo-Alpha_master/commits.json]: keep the sinceDate, untilDate, its will be easier to generate the github link to show just that period

**Pull request merged**